pymeasure/adapters/prologix.py

```diff
@@ -168,13 +175,20 @@ def gpib(self, address, rw_delay=None):
         rw_delay = rw_delay or self.rw_delay
         return PrologixAdapter(self.connection, address, rw_delay=rw_delay)
 
-    def wait_for_srq(self, timeout=25, delay=0.1):
+    def _check_for_srq(self, rw_delay):
+        # it was int(self.ask("++srq"))
+        self.write("++srq")
+        time.sleep(rw_delay or self.rw_delay)
+        return int(self.read())
+
+    def wait_for_srq(self, timeout=25, delay=0.1, rw_delay=0):
```

Comment:
Needing a second `rw_delay` here is I think another friction point generated by `ask` not being in `Adapter` anymore?

Reply:
Delay could also be called interval. I think the cost is appropriate for the benefit to define the read/write logic just once in the Instrument and not in read, write, ask, values

Reply:
Fair enough, I certainly see the benefit, but I also see the friction points it creates ;-)